src/main/java/xyz/nucleoid/plasmid/game/stats/StatisticMap.java

```diff
@@ -0,0 +1,68 @@
+package xyz.nucleoid.plasmid.game.stats;
+
+import com.google.gson.JsonObject;
+import it.unimi.dsi.fastutil.objects.Object2ObjectMap;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+
+/**
+ * Stores a mapping of {@link StatisticKey} to their corresponding values
+ */
+public class StatisticMap {
+    private final Object2ObjectMap<StatisticKey<?>, Number> values = new Object2ObjectOpenHashMap<>();
+
+    public void increment(StatisticKey<Double> key, double amount) {
+        this.values.put(key, (Double) this.values.getOrDefault(key, 0.0d) + amount);
+    }
+
+    public void increment(StatisticKey<Float> key, float amount) {
+        this.values.put(key, (Float) this.values.getOrDefault(key, 0.0f) + amount);
+    }
+
+    public void increment(StatisticKey<Integer> key, int amount) {
+        this.values.put(key, (Integer) this.values.getOrDefault(key, 0) + amount);
+    }
+
+    public double getDouble(StatisticKey<Double> key) {
```

Comment:
Ahh erased generics 😆 
Might be best to name these all consistently with type suffix since we can't avoid it here